Core/src/Core/Filters/Algorithms/WriteASCIIData.cpp

```diff
@@ -0,0 +1,350 @@
+        WriteOutASCIIData<int16>(this, m_DataStructure.getDataRefAs<DataArray<int16>>(selectedArrayPath), maxValPerLine, delimiter, filePath).execute();
+        break;
+      }
+      case DataType::int32: {
+        WriteOutASCIIData<int32>(this, m_DataStructure.getDataRefAs<DataArray<int32>>(selectedArrayPath), maxValPerLine, delimiter, filePath).execute();
+        break;
+      }
+      case DataType::int64: {
+        WriteOutASCIIData<int64>(this, m_DataStructure.getDataRefAs<DataArray<int64>>(selectedArrayPath), maxValPerLine, delimiter, filePath).execute();
+        break;
+      }
+      case DataType::uint8: {
+        WriteOutASCIIData<uint8>(this, m_DataStructure.getDataRefAs<DataArray<uint8>>(selectedArrayPath), maxValPerLine, delimiter, filePath).execute();
+        break;
+      }
+      case DataType::uint16: {
+        WriteOutASCIIData<uint16>(this, m_DataStructure.getDataRefAs<DataArray<uint16>>(selectedArrayPath), maxValPerLine, delimiter, filePath).execute();
+        break;
+      }
+      case DataType::uint32: {
+        WriteOutASCIIData<uint32>(this, m_DataStructure.getDataRefAs<DataArray<uint32>>(selectedArrayPath), maxValPerLine, delimiter, filePath).execute();
+        break;
+      }
+      case DataType::uint64: {
+        WriteOutASCIIData<uint64>(this, m_DataStructure.getDataRefAs<DataArray<uint64>>(selectedArrayPath), maxValPerLine, delimiter, filePath).execute();
+        break;
+      }
+      case DataType::float32: {
+        WriteOutASCIIData<float32>(this, m_DataStructure.getDataRefAs<DataArray<float32>>(selectedArrayPath), maxValPerLine, delimiter, filePath).execute();
+        break;
+      }
+      case DataType::float64: {
+        WriteOutASCIIData<float64>(this, m_DataStructure.getDataRefAs<DataArray<float64>>(selectedArrayPath), maxValPerLine, delimiter, filePath).execute();
+        break;
+      }
+      default: {
+        throw std::runtime_error("Invalid DataType");
+      }
+      }
+      count++;
+    }
+  }
+  else if(static_cast<WriteASCIIDataFilter::OutputStyle>(m_InputValues->outputStyle) == WriteASCIIDataFilter::OutputStyle::SingleFile) // SingleFile = 1
+  {
+    for(const auto& selectedArrayPath : selectedDataArrayPaths)
+    {
+      if(m_ShouldCancel)
+      {
+        return {};
+      }
+      m_MessageHandler(fmt::format("Generating Header Number: {}", count + 1));
+      auto& selectedArrayPtr = m_DataStructure.getDataRefAs<IDataArray>(selectedArrayPath);
+      if(count == 0) // initialize file path
+      {
+        filePath = getFilePath(selectedArrayPtr);
+      }
+      std::ofstream fout(filePath, std::ios_base::app); // open precreated file in append mode
+      fout << selectedArrayPtr.getName() << delimiter;
+      count++;
+      if(count == selectedDataArrayPaths.size())
+      {
+        fout << "\n";
+      }
+      fout.close();
+    }
+    count = 0; // reset
+    // begin printing arrays
+    for(const auto& selectedArrayPath : selectedDataArrayPaths)
```

Comment:
@nyoungbq As far as I can tell this for loop is the same as the one in the MultipleFiles if-statement above.  If that's true then perhaps we avoid duplicating this code by moving it outside the if statements.

Reply:
@jmarquisbq The only difference was updating the filePath repeatedly if MultipleFiles so I just nested an if in the for loop to update the filePath and removed the repeated code and stripped out a pointless counter. Should be fixed now.

Reply:
@nyoungbq When I looked at that originally I somehow saw that extra code being right before hand outside the for loop.  Either way though that looks good 👍